### PR TITLE
New PID adjustments code size optimization

### PR DESCRIPTION
**Pull request by NightHawk32**
Code for Setting the adjustments is now reused for every adjustment. This saves approx. 1.3K compared to the old solution
I tried different attempts, but this one still remains readable and the performance thouldn't be too bad.
@hydra please have a look at this, maybe this can still be tweaked a little.

### CODE REVIEW

File: src/main/io/rc_controls.c [hunk 1]
```diff
@@ -380,10 +380,18 @@ void configureAdjustmentState(adjustmentRange_t *adjustmentRange) {
 
     MARK_ADJUSTMENT_FUNCTION_AS_READY(index);
 }
+  
+void setAdjustment(uint8_t* ptr, float* ptrFP, uint8_t adjustment, int delta, float factor, bool canBeFp, uint8_t min, uint8_t max) {
```
Comment (ledvinap) [inline src/main/io/rc_controls.c]: `canBeFp` can be replaced by `ptrFP ~= NULL`

File: src/main/io/rc_controls.c [hunk 2]
```diff
@@ -392,260 +400,126 @@ void applyStepAdjustment(controlRateConfig_t *controlRateConfig, uint8_t adjustm
     }
     switch(adjustmentFunction) {
```
Comment (ledvinap) [inline src/main/io/rc_controls.c]: this switch can now be replaced by table, with small switch to handle updating of curves ...

File: src/main/io/rc_controls.c (hunk 1 above)
Comment (NightHawk32) [inline src/main/io/rc_controls.c]: Thanks for the catch, removed that parameter.

File: src/main/io/rc_controls.c (hunk 2 above)
Comment (NightHawk32) [inline src/main/io/rc_controls.c]: I spent some thoughts about making a table but that would require collecting all the pointers at startup and there will be some confusion about the adjustments which change two values at time, so there would also offsets be needed to find the right row.

Comment (ledvinap) [inline src/main/io/rc_controls.c]: some way to decide what area to adjust will be necessary .. should be quite simple, only 2 starting pointers are used now

Comment (ledvinap) [inline src/main/io/rc_controls.c]: `offsetof()` can help with address part ...
dual adjustments are a bit harder ... maybe add new type (ADJUSTMENT_MODE_MULTI) to adjustmentConfig_t and pass individual adjustments indexes in data part ...
Also applyStepAdjustment table may be sparse to save unused functions ..

**Comment by hydra** [comment]
this is definately better than what was there before.  merging.  we can change to a table later as required.  PR's welcomed.

**Comment by hydra** [comment]
merged to master. good code size reduction and less overall logic.

new:
```
arm-none-eabi-size ./obj/main/cleanflight_NAZE.elf
   text    data     bss     dec     hex filename
 122924     272   14516  137712   219f0 ./obj/main/cleanflight_NAZE.elf
arm-none-eabi-objcopy -O ihex --set-start 0x8000000 obj/main/cleanflight_NAZE.elf obj/cleanflight_NAZE.hex
arm-none-eabi-objcopy -O binary obj/main/cleanflight_NAZE.elf obj/cleanflight_NAZE.bin
```

old:
```
arm-none-eabi-size ./obj/main/cleanflight_NAZE.elf
   text    data     bss     dec     hex filename
 123188     272   14508  137968   21af0 ./obj/main/cleanflight_NAZE.elf
arm-none-eabi-objcopy -O ihex --set-start 0x8000000 obj/main/cleanflight_NAZE.elf obj/cleanflight_NAZE.hex
arm-none-eabi-objcopy -O binary obj/main/cleanflight_NAZE.elf obj/cleanflight_NAZE.bin
```

**Pull request merged**